sql/src/main/java/io/crate/planner/consumer/InsertFromSubQueryConsumer.java

```diff
@@ -75,6 +79,15 @@ public PlannedAnalyzedRelation visitInsertFromQuery(InsertFromSubQueryAnalyzedSt
                     statement.tableInfo().isPartitioned()
             );
 
+            InputCreatingVisitor.Context inputContext = new InputCreatingVisitor.Context(statement.columns());
+            for (ReferenceInfo referenceInfo : statement.tableInfo().partitionedByColumns()) {
+                if (!statement.columns().contains(referenceInfo) && referenceInfo instanceof GeneratedReferenceInfo) {
```

Comment:
would it make sense to change the order of the conditions? I think instanceof is usually faster than contains on the columns